modin/engines/base/io/text/text_file_reader.py

```diff
@@ -81,3 +51,223 @@ def pathlib_or_pypath(cls, filepath_or_buffer):
         except ImportError:  # pragma: no cover
             pass
         return False
+
+    @classmethod
+    def offset(
+        cls,
+        f,
+        nrows=None,
+        skiprows=None,
+        chunk_size_bytes=None,
+        quotechar=b'"',
+        is_quoting=True,
+    ):
+        """
+        Moves the file offset at the specified amount of bytes/rows.
+
+        Parameters
+        ----------
+            f: file object
+            nrows: int, number of rows to read. Optional, if not specified will only
+                consider `chunk_size_bytes` parameter.
+            chunk_size_bytes: int, Will read new rows while file pointer
+                is less than `chunk_size_bytes`. Optional, if not specified will only
+                consider `nrows` parameter, if both not specified will read till
+                the end of the file.
+            skiprows: array or callable (optional), specifies rows to skip
+            quotechar: char that indicates quote in a file
+                (optional, by default it's '\"')
+            is_quoting: bool, Whether or not to consider quotes
+                (optional, by default it's `True`)
+
+        Returns
+        -------
+            bool: If file pointer reached the end of the file, but did not find
+            closing quote returns `False`. `True` in any other case.
+        """
+        if nrows is not None or skiprows is not None:
+            return cls._read_rows(
+                f,
+                nrows=nrows,
+                skiprows=skiprows,
+                quotechar=quotechar,
+                is_quoting=is_quoting,
+                max_bytes=chunk_size_bytes,
+            )[0]
+
+        if chunk_size_bytes is None:
+            chunk_size_bytes = -1
```

Comment:
We will reach that line only in that case, when we make a call like that `cls.offset(f, nrows=None, chunk_size_bytes=None)`, but we're never doing that, because we always know either `chunk_size` or number of rows that we want to read

Should we add a test, that testing an internal API?

Reply:
If we do not test this case (besides, it does not make sense), then it is better to explicitly show that the function is not intended to work where all parameters: `nrows`, `chunk_size_bytes` and `skiprows` are None. Maybe explicit assert?

Reply:
function can handle case when all parameters are `None`, then it will just read till the end of a file, so I don't think we want an assert here

Reply:
It can, but will it ever do?

Reply:
I do not like the idea of adding internal tests for a case that does not and will not happen.

Reply:
> It can, but will it ever do?

If we will use that function somewhere else in the future, then maybe

Reply:
You can add tests later when needed, but for now, make an assert with a message that such a case should not appear in the current conditions, so it has not been tested.

Reply:
done